DietPi-Banner | Support for custom colour codes

Fourdee opened the pull request:
### Status: Ready

- Adds user changeable colour support via: https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475659997
- Improves visibility of secondary items (previously grey, now white with descriptions in bold white)

![image](https://user-images.githubusercontent.com/7077026/54832918-05b43b00-4cb5-11e9-96c1-38bf889a39dd.png)

Commit:
v6.22

+ Adds colour support for DietPi-Banner

Fourdee commented:
Users can then tweak colours set in ```/DietPi/dietpi/.dietpi-banner```
```
aCOLOUR[0]='\e[38;5;154m' #DietPi Green
aCOLOUR[1]='\e[1m' #Bold White | Main descriptions
aCOLOUR[2]='\e[90m' #Grey | Credits
aCOLOUR[3]='\e[91m' #Red | Update notifications
```

Commit:
v6.22

+ Minor, line up seperator to credits.

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -48,7 +48,19 @@
-	GREEN_LINE=' \e[38;5;154m─────────────────────────────────────────────────────\e[0m'
+	aCOLOUR=(
+
+		'\e[0m' #Reset
```
@Fourdee 
Perhaps `\e[0m` should stay hard corded to not allow accidental colour mess?

MichaIng commented:
@Fourdee 
Nice. I already had the same idea with default white for the values, since the grey is very dark on some displays.

Do we want to add some GUI to adjust these colours? But perhaps this can wait for 6.23.

#### Other idea for v6.23:
To finalise DietPi colour customization we could make the colour codes we use globals. E.g. give them reasonable names like "failure", "success", "warning", "info" (G_DIETPI_NOTIFY), "accent" (DietPi green), "highlight" (bold white) and "announcement" (orange). These can be then used throughout our scripts to have a consistent colour theme. Warning and announcement could be also the same colour, not sure.
This also solves some possible issues where not all colour codes are supported by all terminal types, respectively they lead to different results. E.g. on VirtualBox terminal some colouors are quite funny (will post some picture later).

Commit:
v6.22

+ Hard code colour reset
+ Line up main banner
https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475671016

Fourdee commented:
@MichaIng 

> Do we want to add some GUI to adjust these colours? But perhaps this can wait for 6.23.

Yep agree 👍 

> Perhaps \e[0m should stay hard corded to not allow accidental colour mess?

Good call, done.

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -168,11 +168,11 @@ $GREEN_LINE"
- ${aCOLOUR[2]}dietpi-config${aCOLOUR[0]}    $GREEN_SEPERATOR Feature rich configuration tool for your device.
- ${aCOLOUR[2]}dietpi-software${aCOLOUR[0]}  $GREEN_SEPERATOR Select optimized software for installation.
- ${aCOLOUR[2]}htop${aCOLOUR[0]}             $GREEN_SEPERATOR Resource monitor.
- ${aCOLOUR[2]}cpu${aCOLOUR[0]}              $GREEN_SEPERATOR Shows CPU information and stats.\n"
```
👍 Just had the same idea.

Fourdee commented:
@MichaIng 

Hmm, slight issue with header, it will never line up as could be master/beta/dev + 2 digit versions:
![image](https://user-images.githubusercontent.com/7077026/54835524-28951e00-4cba-11e9-9e25-95c4ece4532c.png)

Only solution I see is to get length of ```SIZE=${#DIETPI_VERSION}```, then set additional blanks for each ```aDESCRIPTION``` entry. But this will impact performance.

Fourdee commented:
@MichaIng 

Also want to add MOTD pulled from our server to v6.23 😄

MichaIng commented on dietpi/func/dietpi-banner:
```diff
@@ -31,36 +31,36 @@
-		'Weather (wttr.in)'
-		'Custom banner entry'
+		'Weather (wttr.in)   '
+		'Custom banner entry '
```
@Fourdee 
Note that this entry is customizable. Perhaps we can fill up the returned value from the input box (below) with white spaces to match 20 characters?

Fourdee replied:
@MichaIng 

Hmmm, personally I believe revert back to no alignment for now due to :
https://github.com/MichaIng/DietPi/pull/2649#issuecomment-475673381

?

EDIT: Done : https://github.com/MichaIng/DietPi/pull/2649/commits/c2dce89e6ee27ce062d5316eb0f94cf857c95d7a but can change if needed.

MichaIng replied:
@Fourdee 
Yeah I think it's okay for now. Space calculation time (e.g. via `$(( 20 - ${#var} ))` ) for this two strings (Custom banner entry as well) should be only marginal compared to the external commands we use to obtain banner values, especially those including www access (weather/WAN IP).

But lets move this fine tuning + performance testing to v6.23

Commit:
v6.22

+ Revert line allignment in main banner due to: additional processing required to line up with header aswell.

MichaIng commented:
@Fourdee 
> Also want to add MOTD pulled from our server to v6.23 😄

Yey was also thinking about this for a long time. We could even use this to inform about current bugs/issues found, based on device and installed software, to reduce duplicate bug reports and provide quick solutions on the fly.
__________
Ready from your point of view? Then I will merge. I go through the open support requests in the forum and in case merge last Beta to be released then tomorrow evening.

MichaIng committed:
v6.22

+ DietPi-Banner | Minor alignment and remove non-required colour reset after banner value obtain commands
